### PR TITLE
Bubble-Sort Algorithm: First Ruby Project

### PR DESCRIPTION
**Pull request by ferbaco86**
On this project, we applied all the knowledge we have acquired reading the introductory material on the Ruby section. This is a basic project that required to create a bubble sort algorithm from scratch. This project includes the use of loops, arrays, method definitions, and array methods.
## Project milestones
- Create our first ruby project.
- Apply problem-solving skills, dividing the whole problem into smaller parts.

## General requirements:
- [x] Add a descriptive README file to your repositories as described here.
- [x] Use meaningful names for your Pull Requests and add a short summary of its content as described here.
- [x] Use descriptive commit messages as described here.
- [x] Do not commit files that are not related to the project. Instead, add them to .gitignore file.
- [x] Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.
## Github Flow:
- [x] Use a specific branch for each feature.
- [x] Use a descriptive name for your branches.
- [x] Use Rubocop to validate your code.
## Project-specific
- [x] Build a bubble-sort algorithm.

### CODE REVIEW

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far , but there are a few more things required. 
### General
- [ ] This is a collaborative project but am seeing commits just from one individual.
- [ ] `bubble_sort_by` method should be implemented

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by ferbaco86** [comment]
Hi, thanks for the review.
Would you please clarify where are the requirements for the bubble_sort_by method?
We followed the link on the Pathwright https://www.theodinproject.com/courses/ruby-programming/lessons/bubble-sort and there is no mention to build a bubble_sort_by method
Kind regards

**Comment by lmaldonadoch** [comment]
Hi! Thank you for your review. These were the comments and the solutions:

- [x] This is a collaborative project but I am seeing commits just from one individual.
> Done! Thank you for pointing it out. The first part of the project took us a surprisingly small amount of time, so we didn't push several commits but now there are some from the both of us.

- [x] `bubble_sort_by` method should be implemented
> Done! Thank you for letting us know. [The Odin Project Bubble Sort Project](https://www.theodinproject.com/courses/ruby-programming/lessons/bubble-sort#student-solutions) was updated, and the second part is no longer required there, but we took a look at the archives and found what the bubble_sort_by method was, and we implemented it.

Happy coding!

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      next unless arr[i] > arr[i + 1]
+
+      temp = arr[i + 1]
+      arr[i + 1] = arr[i]
+      arr[i] = temp
+      ordered += 1
+    end
+    cont += 1
+  end
+  p arr
```
Comment (shshamim63) [inline bubble_sort.rb:16]: the method returns a sorted array. It does not print the array.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      next unless yield(arr[i], arr[i + 1]).positive?
+
+      temp = arr[i + 1]
+      arr[i + 1] = arr[i]
+      arr[i] = temp
+      ordered += 1
+    end
+    cont += 1
+  end
+  p arr
```
Comment (shshamim63) [inline bubble_sort.rb:34]: Same here. The method returns a sorted array. It does not print the array.

**Review by shshamim63 — APPROVED** [review]
## Status :clubs: Approved :clubs: 
___
Hi @ferbaco86 and @lmaldonadoch :wave: ,
Congrats, your methods are working correctly, the method bubble_sort_by is not in the Odin Project now. Thanks for informing that. Just got two suggestions to make. Check out the comments below. This :+1: is from me. you can go ahead.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

**Pull request merged**